src/main/ngapp/accio/src/app/editor/magic-wand.service.ts

```diff
@@ -0,0 +1,159 @@
+import { Injectable } from '@angular/core';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class MagicWandService {
+  /** Return set of coordinates(formatted as a 1-D array).
+   *  Coordinates correspond to pixels considered part of the mask.
+   */
+  /* tslint:disable */
+  floodfill(imgData: ImageData, xCoord: number, yCoord: number,
+            tolerance: number): Set<number> {
+    // Use basic queue to immitate recursive approach of the stack
+    // possible TODO: set array size after mvp to optimize heap memory access
+    const visit: Array<Array<number>> = new Array();
+    const visited: Set<number> = new Set();
+    // Use a set for mask; mainly do iter and set operations on masks
+    const mask: Set<number> = new Set();
+    // Represent [R,G,B,A] attributes of initial pixel
+    const originalPixel: Array<number> = this.dataArrayToRGBA(imgData, xCoord, yCoord);
+
+    visit.push([xCoord, yCoord]);
+    // Convert [x,y] format coord to 1-D equivalent of imgData.data (DataArray)
+    let indexAsDataArray: number = this.coordToDataArrayIndicies(xCoord, yCoord, imgData.width)[0];
+    visited.add(indexAsDataArray);
+    let visitSize: number = 1;
```

Comment:
I think sets have a size property? visited.size, is there a reason to store the size separate?

Reply:
I was thinking about the future. When coming back to optimize after we finish the mvp, an idea I had was to declare the array with a bigger default initialized size. If we decide to do this in the future, the .size property wouldn't work/accurately reflect what we want it to anymore.

Reply:
I would say that this is "future proofing code".  By doing this you are now requiring that whenever someone adds/removes something from the array they need to remember to increment/decrement the visitSize.   It is better to use the .size method because then it will never become out of sync.  If/when you implement your performance optimization then you would be trading off the complexity for performance.  Now you are just increasing the complexity for no gain.

Reply:
I'll go ahead remove the complexity for now then.